market_maker_keeper/uniswapv2_market_maker_keeper.py

```diff
@@ -0,0 +1,379 @@
+
+        parser.add_argument("--accepted-slippage", type=float, default=2,
+                            help="Percentage difference between Uniswap exchange rate and aggregated price"
+                                 "(default: 2)")
+
+        parser.add_argument("--factory-address", type=str,
+                            help="Optional address used to test locally deployed contracts")
+
+        parser.add_argument("--router-address", type=str,
+                            help="Optional address used to test locally deployed contracts")
+
+        parser.add_argument("--initial-delay", type=int, default=10,
+                            help="Initial number of seconds to wait before placing liquidity")
+
+        parser.add_argument("--debug", dest='debug', action='store_true',
+                            help="Enable debug output")
+
+        self.arguments = parser.parse_args(args)
+        setup_logging(self.arguments)
+
+        self.web3 = kwargs['web3'] if 'web3' in kwargs else Web3(HTTPProvider(endpoint_uri=f"{self.arguments.rpc_host}:{self.arguments.rpc_port}",
+                                                                              request_kwargs={"timeout": self.arguments.rpc_timeout}))
+
+        self.web3.eth.defaultAccount = self.arguments.eth_from
+        if 'web3' not in kwargs:
+            register_keys(self.web3, self.arguments.eth_key)
+
+        # TODO: Add a more sophisticated regex for different variants of eth on the exchange 
+        # Record if eth is in pair, so can check which liquidity method needs to be used
+        self.is_eth = 'ETH' in self.pair()
+
+        # Assume token a is always ETH if ETH is in pair to simplify logic
+        token_a_name = 'WETH' if self.is_eth else self.pair().split('-')[0]
+        token_b_name = self.pair().split('-')[1]
+
+        self.reloadable_config = ReloadableConfig(self.arguments.token_config)
+        self._last_config_dict = None
+        self._last_config = None
+        token_config = self.get_token_config().tokens
+        self.token_a = list(filter(lambda token: token.name == token_a_name, token_config))[0]
+        self.token_b = list(filter(lambda token: token.name == token_b_name, token_config))[0]
+
+        self.gas_price = GasPriceFactory().create_gas_price(self.arguments)
+
+        self.accepted_slippage = Wad.from_number(self.arguments.accepted_slippage / 100)
+        self.price_feed = PriceFeedFactory().create_price_feed(self.arguments)
+        self.initial_exchange_rate = self.arguments.initial_exchange_rate
+        self.uniswap_current_exchange_price = Wad.from_number(self.initial_exchange_rate) if self.initial_exchange_rate is not None else self.uniswap.get_exchange_rate()
+        self.testing_feed_price = False
+        self.test_price = Wad.from_number(0)
+
+        # used for local testing
+        if self.arguments.factory_address is None and self.arguments.router_address is None:
```

Comment:
Well we do pass in the args when we are running the integration tests in the pipeline and router / factory have been locally deployed. If local testing is being conducted the keeper will utilize the local contracts, otherwise it falls back on the default values which are the same for every pair and network. This can be changed of course, but what would the benefits be?

Reply:
The address should always be the same for the router, it only varies frequently in the local testing setup, which is why I figured it could be simplified by hardcoding.

Reply:
The address is always the same for Oasis contract (aside from it changing every year) The same goes for 0x smart contracts... I think we shouldn't have it hard coded

Reply:
I will update it to rely on the passed in address